### PR TITLE
Add created Date field to Job

### PR DESCRIPTION
**Pull request by davidxia**
So we know when a job was created.

### CODE REVIEW

File: helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java [hunk 1]
```diff
@@ -190,6 +192,7 @@ public Job(@JsonProperty("id") final JobId id,
              @JsonProperty("metadata") @Nullable final Map<String, String> metadata) {
     this.id = id;
     this.image = image;
+    this.created = new Date();
```
Comment (gimaker) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java]: maybe use a millisecond timestamp to be consistent with the timestamps reported in the task status history?

**Comment by mattnworb** [comment]
I suggest using joda-time's `DateTime` instead, which would allow you to specify the timezone you want the value to be in, as `Date` can have unexpected timezone logic.

Datetime is also easier to test against

File: helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java [hunk 2]
```diff
@@ -246,6 +250,10 @@ public String getHostname() {
     return hostname;
   }
 
+  public Date getCreated() {
+    return created;
+  }
```
Comment (mattnworb) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java]: if Job is meant to be immutable, then this method breaks that, as Date can be altered

Comment (davidxia) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java]: Isn't this a getter?

**Comment by davidxia** [comment]
I considered using Joda time and milliseconds. I used Date to be consistent with the expires field in Job class. DateTime seems more human readable than milliseconds. But if we do that itd be nice to also change expires which might be tricky to be backwards compatible. 

I'm also fine with keeping it as Date. Thoughts?

File: helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java (hunk 2 above)
Comment (gimaker) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java]: `Date` is itself not immutable (you can call e.g. `setHour()` on `Date`), so even if the reference remains unchanged this effectively makes `Job` mutable.

**Comment by davidxia** [comment]
I made `created` optional and to be set by the caller. Otherwise the tests become a pain to fix.

**Comment by mattnworb** [comment]
A few other thoughts on this: 

1. I'm not sure if we really care about the JSON format of a job but I think that `"Tue Oct 06 16:02:46 EDT 2015"` is a really ugly format for dates. I think it would be simpler to store a date as milliseconds since epoch, *UTC*, and relying on `Date.toString()` (if that is what is happening here) means that you might get timezone ids in the string based on where the user created the job from, which feels irrelevant. If we don't store as milliseconds then ISO-8601 in utc time still feels better.

2. Can the master assigned the created time to the job instead of generating it on the CLI side? The latter gives us the time on the CLI host, whereas the time on the master feels more authoritative.

**Comment by davidxia** [comment]
@mattnworb Good points, I'll do milliseconds and have the server set it. But we'll need to ignore date when generating the job hash right?

File: helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java [hunk 3]
```diff
@@ -480,6 +498,7 @@ public Builder(final String hash, final Parameters parameters) {
       public String version;
       public String image;
       public String hostname;
+      public Long created;
```
Comment (rohansingh) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java]: Also add something to clear this out in `withoutMetaParameters` below. That keeps it from affecting the hash.

**Comment by rohansingh** [comment]
Milliseconds since epoch is also what we use for the timestamp in task status events. :+1: for consistency.

File: helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java (hunk 3 above)
Comment (davidxia) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java]: Thanks. I was wondering whether it belonged there.

**Comment by mattnworb** [comment]
+1 on the changes in 91262d6, although it feels like it should use the `Clock` abstraction instead of `System.currentTimeMillis()` (but this feels very minor)

**Comment by davidxia** [comment]
@mattnworb yea, it might make testing easier in the future. See latest commit

**Comment by mattnworb** [comment]
I was assuming that the thing that created the JobsResource had a `Clock` instance it would pass in, but :+1: either way

**Comment by gimaker** [comment]
:+1:

**Pull request merged**